download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/cache/Index.java

```diff
@@ -0,0 +1,32 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+/**
+ * Cache index.
+ * @author Paul Polishchuk
+ * @since 1.3.1
+ */
+interface Index {
+
+    /**
+     * Adds given path to the index using url parameter as a key.
+     * @param url index key
+     * @param path index value
+     */
+    void put(String url, String path);
```

Comment:
@mickaelistria OK, let's go step by step :)
Here is the method `void put(String url, String path);`
1) You've proposed using `java.net.URI` instead of `String url`
This would imply following consequences:
- creating a new `java.net.URI` object just to store a value without using any method from that object; that looks weird ... 
- handling checked exception on each creation of `java.net.URI`; do we really want that?
2) You've proposed using `java.io.File` instead of `String path`
- the same reason, we are operating with file paths, not with file themselves;
- `Index` does no operations neither with the path, nor with the file; it's just a convenient wrapper for the Map, which is backed by the binary file to store Map's state. It's not a storage of file bodies. Not a cache, just a Map to search for the path where file locally stored by url where that file was downloaded

I could fully agree with you about if the `Index` would need some specific methods from `java.net.URI` or `java.io.File` inside to check/download/read/write/copy. But this will be high probability to violate single responsibility principle in this pure `Index` abstraction :)
(with, I believe, a good reference to evaluate a design)

P.S. I'm not against strong typing at all, just don't like pure wrapper objects, not an OOP approach.
Hope, my points sounds convincing to you, or you can just throw away this PR and we will forget about it :)